3.8.0 broken on FreeBSD

My CI picked up the new 3.8.0, and the FreeBSD jobs [all fail](https://gitlab.com/xen-project/xen-guest-agent/-/jobs/6380152883):
```
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/os_info-3.8.0/src/freebsd/mod.rs:11:19
   |
11 |     let version = uname()
   |                   ^^^^^-- an argument of type `&str` is missing
   |
note: function defined here
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/os_info-3.8.0/src/uname.rs:5:8
   |
5  | pub fn uname(arg: &str) -> Option<String> {
   |        ^^^^^ ---------
help: provide the argument
   |
11 |     let version = uname(/* &str */)
   |                        ~~~~~~~~~~~~
error[E0308]: mismatched types
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/os_info-3.8.0/src/freebsd/mod.rs:42:14
   |
40 |     match uname("-s") {
   |           ----------- this expression has type `Option<String>`
41 |         None => Type::Unknown,
42 |         Some("MidnightBSD") => Type::MidnightBSD,
   |              ^^^^^^^^^^^^^ expected `String`, found `&str`
error[E0308]: mismatched types
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/os_info-3.8.0/src/freebsd/mod.rs:43:14
   |
40 |     match uname("-s") {
   |           ----------- this expression has type `Option<String>`
...
43 |         Some("FreeBSD") => {
   |              ^^^^^^^^^ expected `String`, found `&str`
```

--- a/os_info/src/freebsd/mod.rs
+++ b/os_info/src/freebsd/mod.rs
@@ -8,7 +8,7 @@ use crate::{bitness, uname::uname, Info, Type, Version};
 pub fn current_platform() -> Info {
     trace!("freebsd::current_platform is called");
 
-    let version = uname()
+    let version = uname("-s")
         .map(Version::from_string)
         .unwrap_or_else(|| Version::Unknown);
 
@@ -29,18 +29,9 @@ fn get_os() -> Type {
         None => return Type::Unknown,
     };
 
-    let os = match Command::new("uname").arg("-s").output() {
-        Ok(o) => o,
-        Err(e) => {
-            error!("Failed to invoke 'uname': {:?}", e);
-            return Type::Unknown;
-        }
-    };
-
-    match uname("-s") {
-        None => Type::Unknown,
-        Some("MidnightBSD") => Type::MidnightBSD,
-        Some("FreeBSD") => {
+    match os.as_str() {
+        "MidnightBSD" => Type::MidnightBSD,
+        "FreeBSD" => {
             let check_hardening = match Command::new("/sbin/sysctl")
                 .arg("hardening.version")
                 .output()
@@ -56,7 +47,8 @@ fn get_os() -> Type {
                 Ok(_) => Type::FreeBSD,
                 Err(_) => Type::FreeBSD,
             }
-        }
+        },
+        _ => Type::Unknown
     }
 }
 

